### ISSUE TITLE
Error the first time redirection (>) is used on Windows 

### ISSUE THREAD

**Issue by melund** [xonsh/xonsh#2469]
If I use redirection on windows it gives me the traceback below. The file is created, but it is empty. 

If I run the command again or an other redirection then I get no more errors and the files are OK: 

```bash
snail@sea ~
$ echo "hello world" > test.txt
xonsh: To log full traceback to a file set: $XONSH_TRACEBACK_LOGFILE = <filename>
Traceback (most recent call last):
  File "C:\Users\mel\Anaconda3\lib\site-packages\xonsh\base_shell.py", line 328, in default
    run_compiled_code(code, self.ctx, None, 'single')
  File "C:\Users\mel\Anaconda3\lib\site-packages\xonsh\codecache.py", line 68, in run_compiled_code
    func(code, glb, loc)
  File "<xonsh-code>", line 1, in <module>
  File "C:\Users\mel\Anaconda3\lib\site-packages\xonsh\built_ins.py", line 877, in subproc_captured_hiddenobject
    return run_subproc(cmds, captured='hiddenobject')
  File "C:\Users\mel\Anaconda3\lib\site-packages\xonsh\built_ins.py", line 813, in run_subproc
    command = HiddenCommandPipeline(specs)
  File "C:\Users\mel\Anaconda3\lib\site-packages\xonsh\proc.py", line 1767, in __init__
    proc = spec.run(pipeline_group=pipeline_group)
  File "C:\Users\mel\Anaconda3\lib\site-packages\xonsh\built_ins.py", line 502, in run
    p = self._run_binary(kwargs)
  File "C:\Users\mel\Anaconda3\lib\site-packages\xonsh\built_ins.py", line 512, in _run_binary
    p = self.cls(self.cmd, bufsize=bufsize, **kwargs)
  File "C:\Users\mel\Anaconda3\lib\site-packages\xonsh\proc.py", line 552, in __init__
    os.set_handle_inheritable(stdout.fileno(), False)
OSError: [WinError 6] The handle is invalid
snail@sea ~
$ echo "hello world" > test1.txt
snail@sea ~
$ echo "hello world" > test2.txt
snail@sea ~
$
snail@sea ~
$ xonfig
+------------------+-------------------------------------------------------+
| xonsh            | 0.5.12.dev64                                          |
| Git SHA          | ab477051                                              |
| Commit Date      | Jul 22 23:08:26 2017                                  |
| Python           | 3.6.1                                                 |
| PLY              | 3.9                                                   |
| have readline    | False                                                 |
| prompt toolkit   | 1.0.14                                                |
| shell type       | prompt_toolkit                                        |
| pygments         | 2.2.0                                                 |
| on posix         | False                                                 |
| on linux         | False                                                 |
| on darwin        | False                                                 |
| on windows       | True                                                  |
| on cygwin        | <xonsh.lazyasd.LazyBool object at 0x000001AFEF5B1A20> |
| is superuser     | False                                                 |
| default encoding | utf-8                                                 |
| xonsh encoding   | utf-8                                                 |
| encoding errors  | surrogateescape                                       |
+------------------+-------------------------------------------------------+
```

Any clues? Otherwise, I will dig into it in the weekend.

**Comment by scopatz** [comment]
Woah, this is wild issue.  Thanks for reporting @melund. I have a total guess of a fix, but could you try #2470?

**Comment by scopatz** [comment]
Closed by #2693
